_posts/2017-01-31-git_v_PhpStormu.md

```diff
@@ -0,0 +1,57 @@
+
+##Commit
+Základním kamenem pro práci s gitem je **commit**. Commit je zachycení stavu souborů tak, jak zrovna jsou. Commit jako takový není dál dělitelný, je tou nejmenší jednotkou, s kterou se v gitu potkáš. Commity se řadí za sebou v čase, jeden následuje druhý tak, jak byly postupně pořizovány. 
+
+> Vytvoř si nový soubor `index.php`. V tomto momentě by měl PhpStorm nabídnout přidání souboru do verzovacího systému. Pokud ne, klikneš pravým myšítkem na soubor a z místní nabídky vyber _Git_ > _Add_. Název přidaného souboru se defaultně zbarví z červené do zelené - to značí, že je soubor nově přidaný do verzovacího systému. 
+> Do `index.php` zapiš obligátní pozdrav celému světu, aby si měli dál s čím pracovat. Vložíme tedy třeba
+ 
+ ```php
+ <?php echo('Hello World!');
+ ```
+ 
+> A teď soubor ulož v čase - slangově commitni. Zvol z horního menu _VCS_ > _Commit changes_ (zkratka _Ctrl + K_). Otevře se okno, do kterého v dolní části vlož popis commitu (commit message), třeba `první commit`. Okno potvrď a je to! 
+
+Správné vymyšlení názvu a popisu commitu (commit message) je celkem náročné. Jsou to totiž zásadní informace, podle kterých se potom v commitech dá nebo nedá vyznat. Název commitu by měl být výstižný a vyjadřovat, co commit s projektem dělá. Dobré popisy jsou třeba: `added sending request via mail` nebo: `fixed bug #34`. Naopak se vyhni popisům typu: `opraveno` nebo: `další commit`. Název commitu by se měl vejít do 50 znaků, do popisu commitu potom patří zbytek. Podrobnější věci ale nepatří ani tam, protože ty si čtenář může najít přímo v kódu. 
+
+Ve velkých projektech se stejně jako na čistotu kódu dbá i na čistotu commitích názvů a jejich popisů. Všechny commity tam jsou popsány stejným stylem - jen tak lze udržet v rozsáhlých projektech přehled a pořádek. Tento přístup se hodí dodržovat i v malých a svých projektech - stejně jako správná úprava kódu tě v budoucnosti ušetří od nejednoho bolení hlavy. Osobně popisuji commity pouze v angličtině a vždy začínám slovesem v minulém čase - popisuji tím, co se s projektem stalo, když se commit objevil. Není to ale jediný ani univerzálně správný způsob. Příklady najdeš v [repozitáři tohoto blogu](https://github.com/tomtomklima/tomtomklima.github.io). 
+
+##Vzdálený repozitář
+Repozitářů může být pro jeden projekt i více a tady přichází ke slovu Github. Github slouží jako **vzdálený repozitář** (remote repository), což je výraz pro složku s projektem někde online na internetu. Důležité je, že se automaticky nesynchronizují, jak by někdo mohl čekat. Vypadá to neprakticky, ale má to svoje výhody: díky tomuto triku si můžeme udržovat rozpracovanou práci (klidně verzovanou commity) a nikam ji ještě nenahrávat online, když není hotová, takže ji ještě nikdo nemusí vidět. Práci se vzdáleným repozitářem zabezpečí příkazy **push**, který práci nahraje online a **pull**, který aktuální projekt stáhne a lokálně uloží. 
```

Comment:
`Github slouží jako vzdálený repozitář` -> není fakticky správně, můžeš to zkusit přeformulovat?